Common/cpp/NativeModules/NativeReanimatedModule.cpp

```diff
@@ -217,29 +207,20 @@ void NativeReanimatedModule::scheduleOnUI(
   });
 }
 
-jsi::Value NativeReanimatedModule::makeSynchronizedDataHolder(
+jsi::Value NativeReanimatedModule::executeOnUIRuntimeSync(
     jsi::Runtime &rt,
-    const jsi::Value &initialShareable) {
-  auto dataHolder = std::make_shared<ShareableSynchronizedDataHolder>(
-      runtimeHelper, rt, initialShareable);
-  return dataHolder->getJSValue(rt);
-}
-
-void NativeReanimatedModule::updateDataSynchronously(
-    jsi::Runtime &rt,
-    const jsi::Value &synchronizedDataHolderRef,
-    const jsi::Value &newData) {
-  auto dataHolder = extractShareableOrThrow<ShareableSynchronizedDataHolder>(
-      rt, synchronizedDataHolderRef);
-  dataHolder->set(rt, newData);
-}
-
-jsi::Value NativeReanimatedModule::getDataSynchronously(
-    jsi::Runtime &rt,
-    const jsi::Value &synchronizedDataHolderRef) {
-  auto dataHolder = extractShareableOrThrow<ShareableSynchronizedDataHolder>(
-      rt, synchronizedDataHolderRef);
-  return dataHolder->get(rt);
+    const jsi::Value &worklet) {
+  auto shareableWorklet = extractShareableOrThrow(rt, worklet);
+  assert(
+      shareableWorklet->valueType() == Shareable::WorkletType &&
+      "only worklets can be scheduled to run on UI");
```

Comment:
Can we use `extractShareableOrThrow<ShareableWorklet>` here instead?
```suggestion
  auto shareableWorklet = extractShareableOrThrow<ShareableWorklet>(rt, worklet, "only worklets can be scheduled to run on UI");
```